src/main/java/com/google/devtools/build/remote/client/DockerUtil.java

```diff
@@ -0,0 +1,108 @@
+
+public final class DockerUtil {
+  private static final String CONTAINER_IMAGE_ENTRY_NAME = "container-image";
+  private static final String DOCKER_IMAGE_PREFIX = "docker://";
+
+  /**
+   * Gets uid of the current user. If the uid could not be fetched, prints a message to stderr and
+   * returns -1.
+   */
+  private static long getUid() {
+    ProcessBuilder processBuilder = new ProcessBuilder();
+    processBuilder.command("id", "-u");
+    try {
+      InputStream stdout = processBuilder.start().getInputStream();
+      byte[] output = ByteStreams.toByteArray(stdout);
+      return Long.parseLong(new String(output).trim());
+    } catch (IOException | NumberFormatException e) {
+      System.err.printf(
+          "Could not fetch UID for passing to Docker container. The provided docker "
+              + "command will not specify a uid (error: %s)\n",
+          e.toString());
+      return -1;
+    }
+  }
+
+  /**
+   * Checks Action for Docker container definition.
+   *
+   * @return The docker container for the Action. If not container could be found, returns null.
+   */
+  private static @Nullable String dockerContainer(Action action) {
+    String result = null;
+    for (Platform.Property property : action.getPlatform().getPropertiesList()) {
+      if (property.getName().equals(CONTAINER_IMAGE_ENTRY_NAME)) {
+        if (result != null) {
+          // Multiple container name entries
+          throw new IllegalArgumentException(
+              String.format(
+                  "Multiple entries for %s in action.Platform", CONTAINER_IMAGE_ENTRY_NAME));
+        }
+        result = property.getValue();
+        if (!result.startsWith(DOCKER_IMAGE_PREFIX)) {
+          throw new IllegalArgumentException(
+              String.format(
+                  "%s: Docker images must be stored in gcr.io with an image spec in the form "
+                      + "'docker://gcr.io/{IMAGE_NAME}'",
+                  CONTAINER_IMAGE_ENTRY_NAME));
+        }
+        result = result.substring(DOCKER_IMAGE_PREFIX.length());
+      }
+    }
+    return result;
+  }
+
+  /**
+   * Outputs a Docker command that will execute the given action in the given path.
+   *
+   * @param action The Action to be executed in the output docker container command.
+   * @param command The Command of the Action being executed. This must match the Command that is
+   *     referred to from the input parameter Action.
+   * @param workingPath The path that is to be the working directory that the Action is to be
+   *     executed in.
+   */
+  public static String getDockerCommand(Action action, Command command, String workingPath) {
+    String container = dockerContainer(action);
+    if (container == null) {
+      throw new IllegalArgumentException("No docker image specified in given Action.");
+    }
+    List<String> commandElements = new ArrayList<>();
+    commandElements.add("docker");
+    commandElements.add("run");
+
+    long uid = getUid();
```

Comment:
Let's keep this for now but remind me to ask Seth about this when he's back! (e.g., Foundry bots I think use a different user)